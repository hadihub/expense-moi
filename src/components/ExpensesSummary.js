import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import getExpenses from "../selectors/expenses";
import getExpensesTotal from "../selectors/expensesTotal";

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? "expense" : "expenses";
  const formattedExpenseTotal = numeral(expensesTotal / 100).format("$0,00.00");
  return (
    <div>
      <p>
        Viewing {expenseCount} {expenseWord} totalling {formattedExpenseTotal}
      </p>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = getExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: getExpensesTotal(visibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
