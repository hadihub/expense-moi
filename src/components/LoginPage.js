import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";

const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">AccountAnt App</h1>
      <p>Don't worry you Ant gonna loose track of your expenses</p>
      <button onClick={startLogin}>Login</button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
