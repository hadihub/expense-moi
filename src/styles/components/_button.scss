.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-weight: 300;
  line-height: 1;
  font-size: 1.8rem;
  padding: $s-size;
  text-decoration: none; //remove line of link
}

.button--link {
  background: none;
}
