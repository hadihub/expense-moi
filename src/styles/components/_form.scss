.form {
  display: flex;
  flex-direction: column;
  //target all direct children of form
  > * {
    margin-bottom: $s-size;
  }
}

.from__error {
  margin: 0 0 $m-size 0;
}
