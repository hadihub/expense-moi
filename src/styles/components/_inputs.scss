//! do not use .input to prevent breaking third party components
.text-input {
  border: 1px solid #cacccd;
  height: 50px;
  font-size: 300;
  font-weight: 300;
  padding: $s-size;
}

.select {
  @extend .text-input;
}

.textarea {
  @extend .text-input;
  height: 10rem;
}
