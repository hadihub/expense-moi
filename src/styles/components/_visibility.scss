.show-for-mobile {
  @media (min-width: $desktop-breakpoint) {
    display: none;
  }
}

.show-for-desktop {
  // To make sure we dont have both cases substract 0.01rem
  @media (max-width: $desktop-breakpoint - 0.01rem) {
    display: none;
  }
}
