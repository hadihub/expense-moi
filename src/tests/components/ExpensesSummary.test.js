import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("should render ExpenseSummary with 1 expense", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={500} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should render ExpenseSummary with multiple expenses", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={15} expensesTotal={16500} />
  );
  expect(wrapper).toMatchSnapshot();
});
