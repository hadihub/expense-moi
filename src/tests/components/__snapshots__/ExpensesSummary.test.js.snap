// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing
      <span>
         
        1
         
      </span>
       
      expense
       totalling
       
      <span>
         
        $5.00
         
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing
      <span>
         
        15
         
      </span>
       
      expenses
       totalling
       
      <span>
         
        $165.00
         
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
